### PR TITLE
Testing the compiled file

### PR DESCRIPTION
**Pull request by KennedyTedesco**
As discussed here: https://github.com/ClassPreloader/ClassPreloader/issues/48

This PR is just for testing purpose.

### CODE REVIEW

**Comment by KennedyTedesco** [comment]
@GrahamCampbell Take a look:

https://travis-ci.org/ClassPreloader/ClassPreloader/builds/62860310

```php
Cannot mix bracketed namespace declarations with unbracketed namespace declarations
```

What about an implementation to add brackets in all namespaces?

Like this:

```php
<?php
namespace Illuminate\Contracts\Container {
    use Closure;

    interface Container
    {
        public function bound($abstract);
        public function alias($abstract, $alias);
    }
}

namespace {
    class MyClass
    {
        public function doSomething()
        {

        }
    }
}
```

**Comment by GrahamCampbell** [comment]
Ok, thanks. Feel free to add an implementation to this PR to fix the issue. :)

**Comment by GrahamCampbell** [comment]
Just nudging travis.

**Comment by KennedyTedesco** [comment]
@GrahamCampbell That was the best solution I have found.

I've tested this implementation on a Laravel 5 application and was generated a compiled file with almost 20k of lines. Worked fine.

image

File: src/Command/PreCompileCommand.php [hunk 1]
```diff
@@ -145,12 +146,47 @@ protected function getCode($file)
...
+    /**
+     * Check parsed code for having namespaces.
+     *
+     * @param array $parsed
+     *
+     * @return bool
+     */
+    protected function parsedCodeHasNamespaces(array $parsed)
+    {
+        // Namespaces can be only on first level in the code,
```
Comment (GrahamCampbell) [inline src/Command/PreCompileCommand.php]: "can only be" rather than "can be only" reads better i think

**Comment by GrahamCampbell** [comment]
Thanks, you code looks pretty good. Just two really minor things to fix, then I'll merge this. :)

**Comment by GrahamCampbell** [comment]
Hmmm. Just wondering if this would be done with a visitor. That would be a really clean way of doing it.

**Comment by KennedyTedesco** [comment]
I tried using a visitor, but it didn't work, because I haven't found a way to add bracket on a namespace even doing parsing and was getting too complex.

```php
namespace Foo\Bar;
foo;
```

```
0: Stmt_Namespace(
    name: Name(
        parts: array(
            0: Foo
            1: Bar
        )
    )
    stmts: array(
        0: Expr_ConstFetch(
            name: Name(
                parts: array(
                    0: foo
                )
            )
        )
    )
)
```

```php
namespace Foo\Bar {
    foo;
}
```

```
0: Stmt_Namespace(
    name: Name(
        parts: array(
            0: Foo
            1: Bar
        )
    )
    stmts: array(
        0: Expr_ConstFetch(
            name: Name(
                parts: array(
                    0: foo
                )
            )
        )
    )
)
```

The question is: How to add `{ }` on this node `namespace Foo\Bar;`? I haven't found a way.

If someone have a idea or knows how to do it, feel free. =)

File: src/Command/PreCompileCommand.php (hunk 1 above)
Comment (KennedyTedesco) [inline src/Command/PreCompileCommand.php]: Done! =)

**Comment by GrahamCampbell** [comment]
> This PR is just for testing purpose.

Are you happy that this is complete, and for me to merge this?

**Comment by KennedyTedesco** [comment]
Yes, I finished.

**Comment by GrahamCampbell** [comment]
Thank you @KennedyTedesco. :)

**Comment by GrahamCampbell** [comment]
This will be available using `1.4.*@dev` until I tag the 1.4.0 release.

**Comment by KennedyTedesco** [comment]
Great! Thanks, mate. :beers:

**Pull request merged**